src/sql/workbench/services/query/common/gridDataProvider.ts

```diff
@@ -132,6 +135,42 @@ export async function getResultsString(provider: IGridDataProvider, selection: S
 	return copyString;
 }
 
+export function getTableHeaderString(provider: IGridDataProvider, selection: Slick.Range[]): string {
+	let headers: Map<number, string> = new Map(); // Maps a column index -> header
+
+	selection.forEach((range) => {
+		let startCol = range.fromCell;
+		let columnHeaders = provider.getColumnHeaders(range);
+		if (columnHeaders !== undefined) {
+			let idx = 0;
+			for (let header of columnHeaders) {
+				headers.set(startCol + idx, header);
+				idx++;
+			}
+		}
+	});
+
+	headers = sortMapEntriesLeftToRight(headers)
+
+	const copyString = Array.from(headers.values())
+		.map(colHeader => colHeader ? colHeader : '')
+		.join('\t');
+
+	return copyString;
+}
+
+/**
+ * Ensures that table entries in the map appear in left to right order instead of the order that they were selected.
+ * @param map Contains the entries selected in a table
+ * @returns Sorted map with entries appearing in left to right order.
+ */
+function sortMapEntriesLeftToRight(map: Map<number, any>): Map<number, any> {
```

Comment:
nit: Maybe instead of "left to right" say "column order"? Or even "index order"? Left to right being concepts in code is a bit odd and may be confusing at first glance.